docs/core/extensions/options-validation-generator.md

```diff
@@ -12,98 +12,106 @@ In the [options pattern](options.md), various methods for validating options are
 
 The content that follows takes the annotation attributes example that's shown in [Options pattern](options.md) and converts it to use the options validation source generator.
 
 Consider the following *appsettings.json* file:
 
-```json
-{
-  "MyCustomSettingsSection": {
-    "SiteTitle": "Amazing docs from Awesome people!",
-    "Scale": 10,
-    "VerbosityLevel": 32
-  }
-}
-```
+:::code language="json" source="snippets/configuration/console-validation-gen/appsettings.json":::
 
 The following class binds to the `"MyCustomSettingsSection"` configuration section and applies a couple of `DataAnnotations` rules:
 
-:::code language="csharp" source="snippets/configuration/console-json/SettingsOptions.cs":::
+:::code source="snippets/configuration/console-validation-gen/SettingsOptions.cs":::
 
-In the preceding `SettingsOptions` class, the `ConfigurationSectionName` property contains the name of the configuration section to bind to. In this scenario, the options object provides the name of its configuration section. Then, call the following code to bind the configuration section to the options object and validate the data annotations:
+In the preceding `SettingsOptions` class, the `ConfigurationSectionName` property contains the name of the configuration section to bind to. In this scenario, the options object provides the name of its configuration section. The use of the following data annotation attributes are used:
 
-```csharp
-builder.Services
-    .AddOptions<SettingsOptions>()
-    .Bind(Configuration.GetSection(SettingsOptions.ConfigurationSectionName))
-    .ValidateDataAnnotations();
-```
+- <xref:System.ComponentModel.DataAnnotations.RequiredAttribute>: Specifies that the property is required.
+- <xref:System.ComponentModel.DataAnnotations.RegularExpressionAttribute>: Specifies that the property value must match the specified regular expression pattern.
+- <xref:System.ComponentModel.DataAnnotations.RangeAttribute>: Specifies that the property value must be within a specified range.
 
-While this approach is effective, it can affect performance because validation occurs at run time via reflection. Additionally, reliance on reflection can lead to issues, particularly when using AOT compilation. By leveraging compile-time source generation for options validation, you can generate performance-optimized validation code and eliminate the need for reflection, resulting in smoother AOT-compatible app building. The following code demonstrates how to use the options validation source generator:
+> [!TIP]
+> In addition to the `RequiredAttribute`, the properties also use the [required](../../csharp/language-reference/keywords/required.md) modifier. This helps to ensure that consumers of the options object don't forget to set the property value, although it doesn't relate to the validation source generation feature.
```

Comment:
I'm not sure if I'd change this, but it could be misinterpreted. Because source generators create source that is compiled in the project, the code created by the source generator must also obey the `required` modifier. In other words, if the source generator wrote code that didn't set a `required` property, the project wouldn't compile.

Reply:
I see where you're coming from with that, but I think what we have now is clear. As it's worded now, it should make sense from all contexts.

Reply:
@BillWagner `required` is only enforced when using `new`. The validator receives an existing instance of the class, and so would not be affected by it.